src/mito2/src/memtable/key_values.rs

```diff
@@ -220,7 +271,23 @@ impl SparseReadRowHelper {
     ///
     /// # Panics
     /// Time index column must exist.
-    fn new(metadata: &RegionMetadata, rows: &Rows) -> SparseReadRowHelper {
+    fn new(
+        metadata: &RegionMetadata,
+        rows: &Rows,
+        primary_key_encoding: PrimaryKeyEncoding,
+    ) -> SparseReadRowHelper {
+        if primary_key_encoding == PrimaryKeyEncoding::Sparse {
+            let indices = rows
+                .schema
+                .iter()
+                .enumerate()
+                .map(|(index, _)| Some(index))
+                .collect();
```

Comment:
Can we ensure that the positions of time index and fields are fixed so we don't need to reorder them?

Reply:
Left a comment here?